statsd/timer.go

```diff
@@ -45,18 +46,25 @@ type timerFinisher struct {
 	ctx       context.Context
 }
 
-func (t *timerFinisher) Finish() {
-	t.timer.Duration(t.ctx, time.Since(t.startTime), t.tags...)
+func (t *timerFinisher) Finish(ts ...Tags) {
+	tags := t.tags
+	tags = append(tags, ts...)
+	t.timer.Duration(t.ctx, time.Since(t.startTime), tags...)
 }
 
-func (t *timerFinisher) SuccessFinish(errp *error) {
-	tags := append([]Tags{}, t.tags...) // Copy the slice
+func (t *timerFinisher) SuccessFinish(errp *error, ts ...Tags) {
+	tags := t.tags
```

Comment:
Any difference between this and `append([]Tags{}, t.tags...)`?

Reply:
The `append` version copies the slice "header" as well as the values.
The assignation version only copies the slice "header".

In this case this is fine since the subsequent code only mutates the header.

But I kept the original form, just in case.